src/FusedChannelData.ts

```diff
@@ -198,110 +134,37 @@ export default class FusedChannelData {
-      this.channelsDataToFuse = channels;
-      this.fuseMethodRequested = fuseMethod;
-      return;
-    }
-
-    // We will break up the image into one piece for each web-worker
-    // can we assert that npx is a perfect multiple of workersCount??
-    this.fuseWorkersWorking = 0;
-    this.isFusing = true;
-    //  var segmentLength = npx / workersCount; // This is the length of array sent to the worker
-    //  var blockSize = this.height / workersCount; // Height of the picture chunk for every worker
-
-    // Launching every worker
-    //console.log("BEGIN FUSE");
-    for (let index = 0; index < this.workersCount; index++) {
-      this.workers[index].postMessage({
-        msgtype: "fuse",
-        combination: combination,
-        fuseMethod: fuseMethod,
-      });
-    }
+    this.fuseRequested = combination;
+    this.channelsDataToFuse = channels;
   }
 
-  // sum over [{chIndex, rgbColor}]
-  private singleThreadedFuse(combination, channels): void {
-    //console.log("BEGIN");
-
-    // explore some faster ways to fuse here...
-
-    let ar, ag, ab, aa, c, lr, lg, lb, la, opacity, channeldata;
-    let cx, fx, idx;
-    const cl = combination.length;
-
-    const npx4 = this.height * this.width * 4;
-    const npx = this.height * this.width;
-
-    const fused = this.fused;
-    // init the rgba image
-    for (let x = 0; x < npx4; x += 4) {
-      fused[x + 0] = 0;
-      fused[x + 1] = 0;
-      fused[x + 2] = 0;
-      fused[x + 3] = 255;
+  public gpuFuse(renderer: WebGLRenderer): void {
+    const combination = this.fuseRequested;
+    const channels = this.channelsDataToFuse;
+    if (!combination) {
+      return;
     }
-    let value = 0;
-    for (let i = 0; i < cl; ++i) {
-      c = combination[i];
-      idx = c.chIndex;
-      if (!channels[idx].loaded) {
-        continue;
-      }
-      if (c.rgbColor) {
-        channeldata = channels[idx].imgData.data;
-        for (cx = 0, fx = 0; cx < npx; cx += 1, fx += 4) {
-          value = channeldata[cx];
-          lr = c.lut[value * 4 + 0];
-          lg = c.lut[value * 4 + 1];
-          lb = c.lut[value * 4 + 2];
-          la = c.lut[value * 4 + 3];
-          opacity = la / 255.0;
 
-          // what if rgb*opacity > 255?
-          ar = fused[fx + 0];
-          fused[fx + 0] = Math.max(ar, lr * opacity);
-          ag = fused[fx + 1];
-          fused[fx + 1] = Math.max(ag, lg * opacity);
-          ab = fused[fx + 2];
-          fused[fx + 2] = Math.max(ab, lb * opacity);
-          aa = fused[fx + 3];
-          fused[fx + 3] = Math.max(aa, la);
-        }
+    // webgl draw one mesh per channel to fuse.  clear texture to 0,0,0,0
+    this.fuseScene.clear();
+    for (let i = 0; i < combination.length; ++i) {
+      if (combination[i].rgbColor) {
+        const chIndex = combination[i].chIndex;
+        // add a draw call per channel here.
+        // TODO create these at channel creation time!
+        const mat = this.fuseMaterial.clone();
```

Comment:
these clone operations are relatively cheap and don't affect runtime performance in any nontrivial way but still could be optimized away by pre-creating them